x-pack/test/api_integration/apis/metrics_ui/log_entries.ts

```diff
@@ -126,6 +126,55 @@ export default function ({ getService }: FtrProviderContext) {
           expect(messageColumn.message.length).to.be.greaterThan(0);
         });
 
+        it('Returns custom column configurations', async () => {
+          const customColumns = [
+            { timestampColumn: { id: uuidv4() } },
+            { fieldColumn: { id: uuidv4(), field: 'host.name' } },
+            { fieldColumn: { id: uuidv4(), field: 'event.dataset' } },
+            { messageColumn: { id: uuidv4() } },
+          ];
+
+          const { body } = await supertest
+            .post(LOG_ENTRIES_PATH)
+            .set(COMMON_HEADERS)
+            .send(
+              logEntriesRequestRT.encode({
+                sourceId: 'default',
+                startTimestamp: EARLIEST_KEY_WITH_DATA.time,
+                endTimestamp: LATEST_KEY_WITH_DATA.time,
+                center: KEY_WITHIN_DATA_RANGE,
+                columns: customColumns,
+              })
+            )
+            .expect(200);
+
+          const logEntriesResponse = pipe(
+            logEntriesResponseRT.decode(body),
+            fold(throwErrors(createPlainError), identity)
+          );
+
+          const entries = logEntriesResponse.data.entries;
+          const entry = entries[0];
+          expect(entry.columns).to.have.length(4);
+
+          const timestampColumn = entry.columns[0] as LogTimestampColumn;
```

Comment:
Out of curiosity, what impact do these type casts have here?

Reply:
I'm not sure. I copy-pasted this from a test below 😬.

My guess is that the test has no way to know what type of column it is without a type assertion, so the type checker will complain when we check for the value of specific properties that are not present in all possible column types, like `expect(eventDatasetColumn.field)`.

Reply:
Right, no idea why I didn't notice these property access expressions :+1: